### ISSUE TITLE
Feature: Support native ESM

### ISSUE THREAD

**Issue by rschristian** [testing-library/preact-testing-library#36]
- `preact-testing-library` version: `2.0.1`
- `preact` version: `10.5.13`
- `node` version: `14.16.0`
- `npm` (or `yarn`) version: yarn `1.22.10`

**Relevant code or config**

`import { render, screen } from '@testing-library/preact';`

**What you did:**

Tried to import from `@testing-library/preact'`

**What happened:**

```
SyntaxError: The requested module '@testing-library/preact' does not provide an export named 'render'

      at jasmine2 (node_modules/jest-jasmine2/build/index.js:228:5)
```

Problem description:

Currently `@testing-library/preact` does not offer ESM output, which makes it unsuitable for testing in a native ESM environment. This will begin to be a problem especially now that Node 10 is EOL (`"type": "module"` was first valid in v12.0.0, the now oldest support version of Node).

~~[`@testing-library/react` does however support ESM](https://github.com/testing-library/react-testing-library/blob/58150b9efdd5515ad88e2d3373d926ee89093278/package.json#L7), so `preact` here is behind.~~ Edit: Apparently not. Apparently Jest doesn't even follow the widely used practice of the `module` key: https://github.com/facebook/jest/issues/2702#issuecomment-585595612

Suggested solution:

Offer ESM output. Will just take an alteration of the build script.

**Comment by connorjclark** [comment]
Thanks @JoviDeCroock !

FYI I couldn't get an import to work until I put `{"type": "module"}` package.json in `dist/esm`.

**Comment by JoviDeCroock** [comment]
@connorjclark fixed in #58
